src/main/java/org/broadinstitute/hellbender/utils/locusiterator/LocusIteratorByState.java

```diff
@@ -315,12 +315,16 @@ private void lazyLoadNextAlignmentContext() {
             readStates.collectPendingReads();
 
             final Locatable location = getLocation();
-            final Map<String, ReadPileup> fullPileupPerSample = new LinkedHashMap<>();
+
+            // We don't need to keep the pileup elements separated by sample within this method,
+            // since they are just going to get combined into one monolithic pileup anyway
+            // when we construct the final ReadPileup below. This optimization speeds up the
+            // HaplotypeCaller by quite a bit!
```

Comment:
Maybe this will be important for #2321, which relies on the separation of the stratified pileup here. Maybe a new option for the LIBS will be good for improving performance...